src/tx/applyTx/utils.js

```diff
@@ -30,22 +39,26 @@ const checkInsAndOuts = (tx, state, bridgeState, unspentFilter) => {
     ...Object.keys(insValues),
     ...Object.keys(outsValues),
   ]).map(Number);
-  const minGasPrice = bridgeState.minGasPrice || 0;
+  const minGasPrice = BigInt(bridgeState.minGasPrice || 0);
   const gas = Math.max(0, tx.outputs.length * 20000 - tx.inputs.length * 10000);
   for (const color of colors) {
+    const inputValue = insValues[color] || BigInt(0);
+    const outputValue = outsValues[color] || BigInt(0);
     if (
       color === 0 &&
       gas > 0 &&
-      minGasPrice > 0 &&
+      greaterThan(minGasPrice, 0) &&
       tx.type === Type.TRANSFER
     ) {
-      const inputValue = insValues[color];
-      const outputValue = outsValues[color];
-      const txGasPrice = (inputValue - outputValue) / gas;
-      if (txGasPrice < minGasPrice) {
+      const txGasPrice = divide(subtract(inputValue, outputValue), BigInt(gas));
+      if (lessThan(txGasPrice, minGasPrice)) {
         throw new Error(`Tx underpriced`);
       }
-    } else if (!isEqual(insValues[color], outsValues[color])) {
```

Comment:
Fixed a probable mistake here: it wasn't checking that inputs <= outputs for ERC20

Reply:
added test as well 👇